### DISCUSSION TITLE
IntelliJ IDEA copilot plugin fails to update or delete

### DISCUSSION THREAD

**Discussion by simonl65** [community/community#17298] [Copilot Conversations] [answered]
When attempting copilot update, via settings > plugins > Update I get the following error dialogue upon IDE restart:

The IDE failed to install or update some plugins.
Please try again, and if the problem persists, please report it
to https://jb.gg/ide/critical-startup-errors

The cause: java.nio.file.AccessDeniedException: C:\Users\<myUsername>\AppData\Local\JetBrains\Toolbox\apps\IDEA-U\ch-0\213.6777.52.plugins\github-copilot-intellij\copilot-agent\bin\copilot-agent-win.exe

**Comment by simonl65** [comment]
After several IDE restarts I have been able to uninstall CoPilot and re-install the latest version.

**Comment by jansorg** [comment, marked as answer]
Most likely, this was caused by a bug, which did not properly terminate the Copilot helper process when the IDE terminated. The latest release fixed this problem. Restarting Windows to terminate the process is a quick fix to make it apply the update at next restart of the IDE.

**Reply by Remzi1993** [reply]
This issue is still popping up. Most likely a regression. Time to investigate because it's really annoying.

**Reply by princelyroyan** [reply]
Yes got this issue again in the new version update to 1.5.8.5775.. Any solution please?

**Reply by princelyroyan** [reply]
Workaround suggested by @Juraj-Masiar works fine :)

**Reply by veda-tschoening** [reply]
Looks like there's nothing which can be done to reliably prevent this issue currently. Please vote on the following issue. No idea why they talking about that this problem was fixed when looking at the open issue, which clearly says it can't be fixed reliably.

https://youtrack.jetbrains.com/issue/IJPL-148314/Provide-API-for-satellite-process-termination-during-IDE-shutdown

**Reply by kamronbatman** [reply]
The best workaround is to NOT restart when it asks. Shut down the IDE completely. Shut down all other jetbrains processes (toolbox doesn't need to be shut down). Then start the IDE again entirely.

**Reply by lostpebble** [reply]
> The best workaround is to NOT restart when it asks. Shut down the IDE completely. Shut down all other jetbrains processes (toolbox doesn't need to be shut down). Then start the IDE again entirely.

This worked for me, thanks. This issue has been so annoying. Please can the team prioritize a fix for this.

**Reply by titamrtn** [reply]
> The best workaround is to NOT restart when it asks. Shut down the IDE completely. Shut down all other jetbrains processes (toolbox doesn't need to be shut down). Then start the IDE again entirely.

This workaround worked for me

**Reply by Chiralajaya** [reply]
Whenever I close the IDE and reopen it copilot is automatically uninstall, then after i was unable to install Copilot

**Reply by marekokorossmann** [reply]
> The best workaround is to NOT restart when it asks. Shut down the IDE completely. Shut down all other jetbrains processes (toolbox doesn't need to be shut down). Then start the IDE again entirely.

Worked for me also. Thanks! 👍 :-) 
Not needed to restart OS nor reinstall of the Copilot. 
Fixed in less than one minute.

**Reply by mikedpsm** [reply]
Still having this issue in October/2024. The only fix i found was reinstalling intellij

**Reply by Nyde42** [reply]
@mikedpsm I'm also still having this issue. What worked for me is:

- Disable Copilot
- manually close and re-open IDE (don't use the restart button)
- uninstall Copilot
- close and re-open again
- install Copilot
- close and re-open again

I don't know if closing and re-opening is actually necessary after EVERY step, but better to make sure.

**Reply by mikedpsm** [reply]
Yeah, that worked for me too, but as soon as the next update arrived, i had the same issue, so repeating those steps become a pain in the a**

**Comment by OguzhanUmutlu** [comment]
I disabled Copilot, uninstalled and restarted the IDE and installed Copilot again. After these attempts Copilot works just fine.

**Reply by ziadayman179** [reply]
this is really worked thank you

**Comment by meza** [comment]
I'm having the same issue with 1.2.15.2816

**Comment by petervdpas** [comment]
Just disable and update (do not enable on restart)! After update and restart... enable!

**Comment by Juraj-Masiar** [comment]
It just happened to me and the solution was super simple, instead of clicking "Restart idea to install updates", I've closed Webstorm manually and reopened it moments later and all was OK.
I think the build-in "Restart" feature is starting the IDE too soon.

**Reply by nik-tripp** [reply]
This one fixed it for me 👍

**Reply by BlasterM2A** [reply]
For me too, with WebStorm and IDEA. Thx

**Reply by ApricityZ** [reply]
It's useful👍

**Reply by NimaQu** [reply]
works for me

**Reply by palp1tate** [reply]
LGTM!!

**Reply by rni063** [reply]
Worked for me too (Rider 2024.1.2)

**Reply by princelyroyan** [reply]
Works fine :) thank you

**Reply by xzhang001** [reply]
awesome

**Reply by Lunesombre** [reply]
Thanks a lot. Worked for me on IntelliJ 2024.1.1 (Ultimate Edition)

**Reply by samcarton** [reply]
This is the one that seemed to work for me for both Rider and Webstorm, thank you

**Reply by maxoux53** [reply]
Worked for me too, thanks!! (RustRover 2024.1.2)

**Reply by jk5279** [reply]
Worked for me in Pycharm 2024.1.3 version.

**Reply by Gianarci** [reply]
The best workaround. It simply works.

**Reply by Davnn1** [reply]
same issue, same solution here. thanks a lot

**Reply by Wovi10** [reply]
Worked for me in Rider 2024.1.4

**Reply by SamuelClab** [reply]
Worked for me in PhpStorm 2024.1.4

**Reply by dmwyatt** [reply]
The "latest release" has not fixed this problem.

**Reply by Sleepstars** [reply]
Same issue, it works in Pycharm 2024.1.4

**Comment by nik-tripp** [comment]
1.4.19.4986 → 1.5.0.5148 seems to be causing this issue for multiple people today

**Reply by reza-repo** [reply]
This is happening again. preventing me to upgrade to 1.5.

I tried this workaround and it worked: https://github.com/orgs/community/discussions/17298#discussioncomment-8717542

**Comment by KnightPeng** [comment]
I face this problem today, my solution:
- Settings>Plugin>Disable [GitHub Copilot]  
- Restart IDE manually
- Settings>Plugin>Update [GitHub Copilot] 
- Restart IDE with the shown button
- Done!

**Reply by sz332** [reply]
This solved my issue, thank you!

**Comment by Megasteel32** [comment]
running as administrator on windows fixed this issue for me

**Reply by Remzi1993** [reply]
That's not really a fix but a silly workaround. Should work without admin privileges.

**Reply by kamronbatman** [reply]
FYI - It doesn't fix the problem at all. It may fix other problems with similar errors, but not this one.

**Comment by RKlitgaard** [comment]
none of the above fixed it for me, but installing the plugin via the https://plugins.jetbrains.com/plugin/17718-github-copilot/versions worked

Update. still experince that copilot chat is making android studio crash

**Comment by laubrino** [comment]
Same issue here today when updating the plugin.

```
The IDE failed to install or update some plugins.
Please try again, and if the problem persists, please report it
to https://jb.gg/ide/critical-startup-errors

The cause: java.nio.file.AccessDeniedException: C:\Users\...\AppData\Roaming\JetBrains\IntelliJIdea2024.1\plugins\github-copilot-intellij\copilot-agent\bin\copilot-agent-win.exe
```

**Comment by awghdbs** [comment]
My Solution: Update the plugin when no project is open.

1. Disable "Reopen projects on startup" option in:
[File | Settings | Appearance & Behavior | System Settings](jetbrains://PyCharmCore/settings?name=Appearance+%26+Behavior--System+Settings)
2. Restart the IDEA into project list page.
3. Check for updates on this page and update the copilot plugin.

**Reply by hugoattal** [reply]
Thank you so much, works like a charm. Can't wrap my head around why there are SO MANY bugs with this plugin...

**Reply by iwan-soptim** [reply]
Thanks for helping out. Works perfectly.

**Reply by pettysoldier** [reply]
Thank you，it works for me

**Reply by tuguzT** [reply]
This solution worked perfectly for me!

**Comment by seakingii** [comment]
1 Disable Copilot
2 Restart IDE
3 Enable Copilot
4 Upgrade Copilot
5 Restart IDE
6 DONE

**Comment by AntonZelenin** [comment]
For the last several months I've been getting this error every time copilot updates in RustRover on Windows

**Comment by remotemerge** [comment]
I have encountered this error in PHPStorm and WebStorm every time I update the plugin for the past few months. I used to uninstall and reinstall the plugin for a dirty fix...

**Comment by ricar2ce** [comment]
For me, the direct restart of the IDE after the copilot update was the problem.

This fix it for me:

Don't click on "Restart IDE now" after the update. Just close the IDE, wait one minute and start the IDE again.
DONE

**Reply by Remzi1993** [reply]
This worked for me. Thank you very much. I uninstalled it and closed the IDE and installed it again without any errors. 👍

**Reply by tdkhoa95** [reply]
It's worked for me too. Thanks a lot

**Reply by SA2687890878** [reply]
the way is right,thank you

**Comment by Remzi1993** [comment]
There is a regression because this is happening to me too.

**Comment by veda-tschoening** [comment]
Still happening for weeks here as well:

```
The IDE failed to install or update some plugins.
Please try again, and if the problem persists, please report it
to https://jb.gg/ide/critical-startup-errors

The cause: java.nio.file.AccessDeniedException: C:\Users\THS\AppData\Roaming\JetBrains\IntelliJIdea2024.1\plugins\github-copilot-intellij\copilot-agent\bin\copilot-agent-win.exe
```

Looks like there's nothing which can be done to reliably prevent this issue currently. Please vote on the following issue. No idea why they talking about that this problem was fixed when looking at the open issue, whcih clearly says it can't be fixed reliably.

https://youtrack.jetbrains.com/issue/IJPL-148314/Provide-API-for-satellite-process-termination-during-IDE-shutdown

**Reply by JKamsker** [reply]
Same for me

**Comment by sebastiancadena** [comment]
+1

**Comment by iqbalShafiq** [comment]
When updating to 1.5.9.5799 I also experienced the same thing. However, I managed to fix this by not clicking restart when prompted. I closed IntelliJ IDEA after downloading the GitHub Copilot update and reopened it manually.

**Comment by radutomy** [comment]
Happening to me as well on Rust Rover with Copilot Plugin. Windows 11

**Comment by kitswas** [comment]
This still happens with Android Studio.  
I encountered it twice or thrice in the past week.

**Comment by ebesavara** [comment]
Do you have plan to do something with that?! It's happening again in datagrip. in one version it was solved now in newest it's back. 
the hell... Datagrip 2024.1.4  on Windows 11 

![datagrip64_QJYIZffqWY](https://github.com/community/community/assets/130746701/144ec444-24d1-4155-94b8-94347c15d189)

**Comment by thoscilo** [comment]
Issue still observed:

``The cause: java.nio.file.AccessDeniedException: C:\Users\USER_NAME\AppData\Roaming\JetBrains\PyCharmCE2024.1\plugins\github-copilot-intellij\copilot-agent\bin\copilot-agent-win.exe``

Unable to update: `1.5.10.5840 → 1.5.11.5872`

Used version:
PyCharm 2024.1.3 (Community Edition)
Build #PC-241.17890.14, built on June 4, 2024
Runtime version: 17.0.11+1-b1207.24 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 11.0

**Reply by thoscilo** [reply]
Those steps resolved issue as W/A:
- uninstall `GH Copilot`
- restart ide
- install `GH Copilot`

**Comment by AMueckl** [comment]
The recipe @ricar2ce already mentioned above broken down in steps would go like this:

- let the `GH Copilot` update as usual, but do not use the `restart IDE` action of the update process
- close the IDE using the `File` -> `Exit` action manually
- start the IDE again

**Comment by 1592363624** [comment]


**Reply by jwwsjlm** [reply]
这对我好像没效果..更新完成copilot之后.不要立马重启ide...等个1分钟2分钟..然后再手动关闭打开 正常了..

**Reply by jwwsjlm** [reply]
> 这对我的好像没效果..更新完成copilot之后.不要立马重启ide...等个1分钟2分钟..然后再手动关闭打开正常了..

或者任务管理器里面给copilot-agent-win这个进程手动结束...

**Comment by yidasanqian** [comment]
The IDE failed to install or update some plugins.
Please try again, and if the problem persists, please report it
to https://jb.gg/ide/critical-startup-errors

The cause: java.nio.file.AccessDeniedException: C:\Users\xx\AppData\Roaming\JetBrains\IntelliJIdea2023.3\plugins\github-copilot-intellij\copilot-agent\bin\copilot-agent-win.exe

**Comment by lecom86** [comment]
Hello everyone,

I'm also having trouble to update all these AI-related plugins [*1] (same "java.nie.file Exception ... access denied" thing) in the traditional way. But I found this WORKAROUND to be useful (a bit tedious, but considering the amount of time I lost by trying all the different proposals in this thread it's pretty short time):

Steps that worked for me:

- Close your IDE.
- Download the plugin's .ZIP from [plugins.jetbrains.com](url) (stable version or the one you want)
- Delete `C:\Users\<YOUR USER>\AppData\Roaming\JetBrains\<YOUR IDE>\plugins\<YOUR PLUGIN>` folder.
- Extract the .zip into `C:\Users\<YOUR USER>\AppData\Roaming\JetBrains\<YOUR IDE>\plugins`.
- Go up one level in Windows explorer.
- Right click the `C:\Users\<YOUR USER>\AppData\Roaming\JetBrains\<YOUR IDE>\plugins` folder, and hit **properties**.
- Make sure "Read-only" is turned completely OFF and **apply changes for this folder and all subfolders**.
- Apply and close dialog.
- Start your IDE.

Hope that helped.



[*1] AI-related plugins being Github Copilot, TabNine and Codeium.

**Reply by jmbianca** [reply]
It work for the Codeium plugin too! Many thanks!!

**Comment by chimchimp** [comment]
This workaround does work, but so annoying to close all projects to bare metal and shutdown.  It's 2024, why hasn't this bug been fixed???! Github, I want my money back!

**Comment by DeoVermaak** [comment]
What worked for me (though might have just been per chance)

I launched Rider and in the dialog where you select the project you want to open, I instead went to the bottom left menu, selected plugins, clicked the update copilot plugin, and then clicked restart IDE and it restarted correctly for me

**Reply by DanOpcode** [reply]
Thanks, that worked for me too

**Reply by colomborodrigo** [reply]
Me too. Thanks for sharing

**Comment by aliforever** [comment]
I had to kill the process name mentioned in the error message, in my case it was `language_server_windows_x64.exe`. Then restarted IDE and updated the plugin successfully.

**Comment by ebesavara** [comment]
That issue is back. This is starting to be nonsense. And there are several other bugs, but i don't even feel like reporting them if you can't solve one. 
![datagrip64_o0NB7I4qvr](https://github.com/user-attachments/assets/5b9183ef-f12a-43f1-b756-6de63fa0d7f4)

**Comment by mderzi** [comment]
seeing the same issue today

![image](https://github.com/user-attachments/assets/d833ab64-2482-4ae8-95db-bd4998f7b287)


any updates?

**Comment by isaacwein** [comment]
![image](https://github.com/user-attachments/assets/50f0effc-a1fb-4f9b-8f1c-76e6b517c6b4)

i have this issue today with Goland 2024.2.1.1
copilot 1.5.21.6667

i tried to uninstall copilot

**Comment by naefp** [comment]
hit this today...

**Comment by lecom86** [comment]
This workaround keeps working flawlessly for me:

- Update plugins using settings dialog as usual.
- Close settings dialog (and REFUSE to restart IDE when prompted).
- Close IDE manually.
- Delete from `%USERPROFILE%\AppData\Roaming\JetBrains\<IDE_and_VERSION>\plugins` any folders belonging to those specific plugins being updated.
- Once they're fully deleted: Start IDE (these folders will be restored/updated automatically by the IDE at start cause it ends the installation this way)

**Reply by GregSotiropoulos** [reply]
That worked for me, thanks! At first I couldn't delete the copilot folder; it turned out this was due to copilot-language-server.exe running in the background, even after Pycharm was closed. Also there were 3 such processes running. I guess at some point something went wrong and a process could not be killed, so new ones were started upon IDE restarts. After killing them all, I was able to delete the folder, and Pycharm then updated copilot to the latest version.

**Comment by mortensimon** [comment]
Dec 2024, this issue surfaced again! This is the second time (last time happened some months back). Annoying. I modified Noctumsempra's list slightly, trying to keep it short and clear:

* Uninstall plugin - but do NOT restart IDE when prompted
* Close settings and go to File->Exit (manual exit)

At this point, the Copilot-plugin seems to be properly shut down. And removed. You can now re-install it.
